public/app/features/logs/components/log-context/LogRowContextModal.tsx

```diff
@@ -260,22 +293,108 @@ export const LogRowContextModal: React.FunctionComponent<LogRowContextModalProps
     }
   };
 
-  useLayoutEffect(() => {
-    if (!loading && entryElement.current && preEntryElement.current) {
-      preEntryElement.current.scrollIntoView({ block: 'center' });
-      entryElement.current.scrollIntoView({ block: 'center' });
+  const maybeLoadMore = async (place: Place) => {
+    const section = context[place];
+    if (section.loadingState === LoadingState.Loading) {
+      return;
+    }
+
+    setSection(place, (s) => ({
+      ...s,
+      loadingState: LoadingState.Loading,
+    }));
+
+    const currentGen = generationRef.current;
+    try {
+      const newRows = await loadMore(place);
+      if (currentGen === generationRef.current) {
+        setSection(place, (s) => ({
```

Comment:
`s` is something like `currentSection`? I think we can still type that out, WDYT?

Reply:
good point... i went with `section` instead of `currentSection`, i hope that's acceptable (if not, i can change it 😄 )